zeus/optimizer/batch_size/server/batch_size_state/models.py

```diff
@@ -0,0 +1,150 @@
+"""
+Pydantic models for Batch size/Exploration/Measurement/GaussianTsArmState
+"""
+
+from __future__ import annotations
+from typing import Any, Optional
+from uuid import UUID
+
+from pydantic.class_validators import root_validator
+from pydantic.fields import Field
+from pydantic.main import BaseModel
+from zeus.optimizer.batch_size.server.database.schema import (
+    GaussianTsArmState,
+    Measurement,
+    State,
+)
+
+
+class BatchSizeBase(BaseModel):
+    job_id: UUID
+    batch_size: int = Field(gt=0)
+
+    class Config:
+        validate_assignment = True
+        frozen = True
+
+
+class GaussianTsArmStateModel(BatchSizeBase):
+    param_mean: float
+    param_precision: float
+    reward_precision: float
+    num_observations: int = Field(ge=0)
+
+    class Config:
+        orm_mode = True
+        frozen = True
+
+    def to_orm(self) -> GaussianTsArmState:
```

Comment:
I didn't get the rationale of having `to_orm`. I thought repositories query SQLAlchemy for objects that represent a row on the DB table, and then those are turned into Pydantic models with `from_orm`, so `orm_mode = True` is necessary. The question is the other direction. I thought when you want to persist modifications to those rows, you need to make modifications directly on the objects returned by SQLAlchemy for it to internally build the update query, and thus the repository should cache those objects or something. Is this not what's going on?

Reply:
That is for `update` case. But for `create` case, we need to instantiate a new object where we need model -> orm